ql/src/java/org/apache/hadoop/hive/ql/udf/generic/AbstractGenericUDFArrayBase.java

```diff
@@ -49,6 +52,12 @@ protected AbstractGenericUDFArrayBase(String functionName, int minArgCount, int
         this.minArgCount = minArgCount;
         this.maxArgCount = maxArgCount;
         this.outputCategory = outputCategory;
+        this.outputPrimitiveCategory = null;
+    }
+
+    public AbstractGenericUDFArrayBase(String functionName, int minArgCount, int maxArgCount, ObjectInspector.Category outputCategory, PrimitiveObjectInspector.PrimitiveCategory outputPrimitiveCategory) {
```

Comment:
Probably this constructor and variable might not be required in this class.
Please check this comment - https://github.com/apache/hive/pull/3896/files#r1066553884.

Reply:
made changes as recommended